File: client/client/src/main/java/client/util/ManageCollection.java
package client.util;

import model.Personage;

import java.util.concurrent.ConcurrentLinkedDeque;

public class ManageCollection {
    public static int maxPower(ConcurrentLinkedDeque<Personage> heroes) {
        ConcurrentLinkedDeque<Personage> c = new ConcurrentLinkedDeque<Personage>();
        c.addAll(heroes);
        int res = 0;
        while (!c.isEmpty()) {
            int tmp = c.removeFirst().getForce();
            if (tmp > res) {
                res = tmp;
            }
        }
        return res;
    }

    public static int maxHeight(ConcurrentLinkedDeque<Personage> heroes) {
        ConcurrentLinkedDeque<Personage> c = new ConcurrentLinkedDeque<Personage>();
        c.addAll(heroes);
        int res = 0;
        while (!c.isEmpty()) {
            int tmp = c.removeFirst().getHeight();
            if (tmp > res) {
                res = tmp;
            }
        }
        return res;
    }

    public static double maxX(ConcurrentLinkedDeque<Personage> heroes) {
        ConcurrentLinkedDeque<Personage> c = new ConcurrentLinkedDeque<Personage>();
        c.addAll(heroes);
        if (c.isEmpty()){
            return 0;
        }
        double res = Double.MIN_VALUE;
        while (!c.isEmpty()) {
            double tmp = c.removeFirst().getX();
            if (tmp > res) {
                res = tmp;
            }
        }
        return res;
    }

    public static double minX(ConcurrentLinkedDeque<Personage> heroes) {
        ConcurrentLinkedDeque<Personage> c = new ConcurrentLinkedDeque<Personage>();
        c.addAll(heroes);
        if (c.isEmpty()){
            return 0;
        }
        double res = Double.MAX_VALUE;
        while (!c.isEmpty()) {
            double tmp = c.removeFirst().getX();
            if (tmp < res) {
                res = tmp;
            }
        }
        return res;
    }

    public static double maxY(ConcurrentLinkedDeque<Personage> heroes) {
        ConcurrentLinkedDeque<Personage> c = new ConcurrentLinkedDeque<Personage>();
        c.addAll(heroes);
        if (c.isEmpty()){
            return 0;
        }
        double res = Double.MIN_VALUE;
        while (!c.isEmpty()) {
            double tmp = c.removeFirst().getY();
            if (tmp > res) {
                res = tmp;
            }
        }
        return res;
    }

    public static double minY(ConcurrentLinkedDeque<Personage> heroes){
        ConcurrentLinkedDeque<Personage> c = new ConcurrentLinkedDeque<Personage>();
        c.addAll(heroes);
        if (c.isEmpty()){
            return 0;
        }
        double res = Double.MAX_VALUE;
        while (!c.isEmpty()){
            double tmp = c.removeFirst().getY();
            if (tmp < res){
                res = tmp;
            }
        }
        return res;
    }

    public static double minHeight(ConcurrentLinkedDeque<Personage> heroes){
        ConcurrentLinkedDeque<Personage> c = new ConcurrentLinkedDeque<Personage>();
        c.addAll(heroes);
        if (c.isEmpty()){
            return 0;
        }
        double res = Double.MAX_VALUE;
        while (!c.isEmpty()){
            double tmp = c.removeFirst().getHeight();
            if (tmp < res){
                res = tmp;
            }
        }
        return res;
    }
}

File: client/client/src/main/kotlin/model/util/ManageCollection.kt
package client.util

import model.Personage

import java.util.concurrent.ConcurrentLinkedDeque

class ManageCollection() {
    fun maxPower(heroes: ConcurrentLinkedDeque<Personage>): Int {
        val c = ConcurrentLinkedDeque<Personage>()
        c.addAll(heroes)
        var res = 0
        while (!c.isEmpty()) {
            val tmp = c.removeFirst().force
            if (tmp > res) {
                res = tmp
            }
        }
        return res
    }

    fun maxHeight(heroes: ConcurrentLinkedDeque<Personage>): Int {
        val c = ConcurrentLinkedDeque<Personage>()
        c.addAll(heroes)
        var res = 0
        while (!c.isEmpty()) {
            val tmp = c.removeFirst().height
            if (tmp > res) {
                res = tmp
            }
        }
        return res
    }

    fun maxX(heroes: ConcurrentLinkedDeque<Personage>): Double {
        val c = ConcurrentLinkedDeque<Personage>()
        c.addAll(heroes)
        if (c.isEmpty()) {
            return 0.0
        }
        var res = java.lang.Double.MIN_VALUE
        while (!c.isEmpty()) {
            val tmp = c.removeFirst().x
            if (tmp > res) {
                res = tmp
            }
        }
        return res
    }

    fun minX(heroes: ConcurrentLinkedDeque<Personage>): Double {
        val c = ConcurrentLinkedDeque<Personage>()
        c.addAll(heroes)
        if (c.isEmpty()) {
            return 0.0
        }
        var res = java.lang.Double.MAX_VALUE
        while (!c.isEmpty()) {
            val tmp = c.removeFirst().x
            if (tmp < res) {
                res = tmp
            }
        }
        return res
    }

    fun maxY(heroes: ConcurrentLinkedDeque<Personage>): Double {
        val c = ConcurrentLinkedDeque<Personage>()
        c.addAll(heroes)
        if (c.isEmpty()) {
            return 0.0
        }
        var res = java.lang.Double.MIN_VALUE
        while (!c.isEmpty()) {
            val tmp = c.removeFirst().y
            if (tmp > res) {
                res = tmp
            }
        }
        return res
    }

    fun minY(heroes: ConcurrentLinkedDeque<Personage>): Double {
        val c = ConcurrentLinkedDeque<Personage>()
        c.addAll(heroes)
        if (c.isEmpty()) {
            return 0.0
        }
        var res = java.lang.Double.MAX_VALUE
        while (!c.isEmpty()) {
            val tmp = c.removeFirst().y
            if (tmp < res) {
                res = tmp
            }
        }
        return res
    }

    fun minHeight(heroes: ConcurrentLinkedDeque<Personage>): Double {
        val c = ConcurrentLinkedDeque<Personage>()
        c.addAll(heroes)
        if (c.isEmpty()) {
            return 0.0
        }
        var res = java.lang.Double.MAX_VALUE
        while (!c.isEmpty()) {
            val tmp = c.removeFirst().height.toDouble()
            if (tmp < res) {
                res = tmp
            }
        }
        return res
    }
}
